fling/component/client/fedamp_client.py

```diff
@@ -0,0 +1,54 @@
+import copy
+import numpy as np
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+
+from fling.utils.registry_utils import CLIENT_REGISTRY
+from .base_client import BaseClient
+from fling.utils.utils import weight_flatten
+
+@CLIENT_REGISTRY.register('fedamp_client')
+class FedAMPClient(BaseClient):
+    """
+    Overview:
+        This class is the base implementation of client in 'Bold but Cautious: Unlocking the Potential of Personalized
+        Federated Learning through Cautiously Aggressive Collaboration' (FedCAC).
+    """
+
+    def __init__(self, args, client_id, train_dataset, test_dataset=None):
+        """
+        Initializing train dataset, test dataset(for personalized settings).
+        """
+        super(FedAMPClient, self).__init__(args, client_id, train_dataset, test_dataset)
+        self.client_u = copy.deepcopy(self.model)
+
+    def FedAMP_Loss_client(self):
```

Comment:
You can use the ``get_model_difference`` function (defined in fling/utils/torch_utils.py ) for simplification.